python/plugins/processing/algs/qgis/ExportGeometryInfo.py

```diff
@@ -138,7 +138,7 @@ def processAlgorithm(self, parameters, context, feedback):
         self.distance_area = QgsDistanceArea()
         if method == 2:
             self.distance_area.setSourceCrs(source.sourceCrs(), context.transformContext())
-            self.distance_area.setEllipsoid(context.project().ellipsoid())
+            self.distance_area.setEllipsoid(context.ellipsoid())
         elif method == 1:
             coordTransform = QgsCoordinateTransform(source.sourceCrs(), context.project().crs(), context.project())
```

Comment:
Hi @nyalldawson, this algorithm also needs `context.project().crs()` and `context.project()`

`context.project().crs()` is also needed for example by [native:networkanalysis](https://github.com/qgis/QGIS/blob/master/src/analysis/processing/qgsalgorithmnetworkanalysisbase.cpp#L116)

What about some other algs that use the whole context.project() like for example native:rasterize native:loadlayer, native:setprojectvariable, native:stylefromproject, native:bookmarkstolayer, native:categorizeusingstyle, ... ?

Reply:
Yes, it's only a partial fix but addresses the immediate ellipsoid handling need. 

I guess we also need a --project=... argument , and then make sure this is specified if an algorithm requires it. That said, "export geometry info" is a generally badly behaved algorithm and needs a rewrite (in C++). And some of those other algorithms have no real meaning outside of qgis and should be hidden entirely (such as the "select by...." algorithms). I'll add a flag to hide algorithms from the standalone tool to handle this.

Reply:
Thank you very much!

Reply:
Ok, I've updated this PR with the extra logic now...